arch/x86/core/fatal.c

```diff
@@ -232,17 +232,33 @@ EXC_FUNC_NOCODE(IV_MACHINE_CHECK);
 #define SGX	BIT(15)
 
 #ifdef CONFIG_X86_MMU
-static void dump_entry_flags(u32_t flags)
+static void dump_entry_flags(x86_page_entry_data_t flags)
 {
-	printk("0x%03x %s %s %s\n", flags,
-	       flags & MMU_ENTRY_PRESENT ? "Present" : "Non-present",
-	       flags & MMU_ENTRY_WRITE ? "Writable" : "Read-only",
-	       flags & MMU_ENTRY_USER ? "User" : "Supervisor");
+#ifdef CONFIG_X86_PAE_MODE
+	printk("0x%x%x %s, %s, %s, %s\n", (u32_t)(flags>>32),
+	       (u32_t)(flags),
+	       flags & (x86_page_entry_data_t)MMU_ENTRY_PRESENT ?
+	       "Present" : "Non-present",
+	       flags & (x86_page_entry_data_t)MMU_ENTRY_WRITE ?
+	       "Writable" : "Read-only",
+	       flags & (x86_page_entry_data_t)MMU_ENTRY_USER ?
+	       "User" : "Supervisor",
+	       flags & (x86_page_entry_data_t)MMU_ENTRY_EXECUTE_DISABLE ?
```

Comment:
Would it be possible to make this flag a positive one, such as `MMU_ENTRY_EXECUTABLE`?  The reason being that pages should never be writable and executable at the same time, and it's easy to forget to add a flag to disable execution.  I'm asking this because it's a good idea to ensure pages are [either executable or writable, but never both](https://en.wikipedia.org/wiki/W%5EX).

Reply:
So at this level, we are working with the actual flags being expressed in the page table itself, the definition of these defines comes right from the CPU manual. For efficiency I think it's probably best not to do any translation here.

What gets exposed to the kernel and applications for doing this kind of configuration is the memory domain APIs, which have an arch-independent set of defines for this sort of this, that do get translated to the MMU-specific bits in the arch/ code.

Reply:
Yeah, thought so.  It's better to express this restriction at the memory domain level, anyway, so it's consistent across architectures.  Are we enforcing W^X there?

Reply:
I looked at #814 again and @chunlinhan didn't have any defines for setting executable state, but I asked that he add them. They would be added to the K_MEM_PARTITION_* set of defines.